docs/advanced_topics/headless.md

```diff
@@ -0,0 +1,190 @@
+
+#### Rendering on the frontend
+
+Pre-rendering the HTML on the backend may be more convenient if you're happy with the way Wagtail renders it, but it's still difficult to customise the rendering on the frontend. Other headless CMSs provide Rich Text as a sequence of blocks in JSON format. This approach makes it easier to customise the rendering of the blocks without having to find a way to parse the HTML fragment.
+
+This approach is currently the harder approach for managing rich text. A long-term solution for this may involve creating a JS library or plugin.
+
+### 🛑 Multi-Site Support
+
+Multi-site works differently in Headless Wagtail
+
+The notion of a “site” is different for Headless Wagtail. In traditional Wagtail, the domain or port of a Wagtail site are where Wagtail will serve content and the location the end user visits to find the site.
+
+
+But in Headless Wagtail, the domain or port that the end user uses and domain or port that Wagtail serves content on will be different. For example, the end user may visit **www.wagtail.org**, and the website could be a Next.js app that queries a Wagtail instance running at **api.wagtail.org**.
+
+
+Wagtail’s current API implementation will check the host header and port to find the site so that it only returns pages under that site. This means that your site record must be set to **api.wagtail.org**. However, when Wagtail generates URLs, these URLs need to be generated for **www.wagtail.org**.
+
+
+The Wagtail API only allows requests from one site at a time to make sure any site listings are isolated from other sites by default. But the API could be improved in the following ways:
+
+* Allow the site to be specified in the API request.
+* Allow all pages across all sites to be queried on an opt-in basis.
+
+With these approaches, the site record in the Wagtail admin of Headless Wagtail would be set to the domain or port that the end user sees so URLs could be reversed correctly. All API requests would specify the site as a GET parameter.
+
+### 🛑 Form submissions
+
+There’s currently no official API for a headless site to use to submit data to a Wagtail form.
+
+### 🛑 Password-Protected Pages
+
+There currently isn’t a way to view a password-protected page from a headless frontend. The API currently excludes all password-protected pages from queries.
+
+### ⚠️ Internationalisation
+
+TODO
+
+### ⚠️ StreamField
+
+TODO
+
+## Frontend
+
+There are a few options to build your frontend for Wagtail.
+
+### ⚠️ Next.js
+
+Next.js is a [popular open source JavaScript framework](https://nextjs.org/) you can choose for building the frontend of your Headless Wagtail website.
+
+There's no specific support for Next.js in Headless Wagtail. This website you're on right now is built with Headless Wagtail and Next.js, so you can have a look at the source code to get an idea how to set up your own project.
```

Comment:
This needs rewording, the docs site won't be on Next.js.

Reply:
Thanks @lb-, for catching that. I've made a suggestion for rewording this, more in line with the Nuxt paragraph.